spec/system/job_applications/status_transition_spec.rb

```diff
@@ -0,0 +1,365 @@
+
+        #
+        # check tab new
+        #
+        publisher_ats_applications_page.select_tab(:tab_submitted)
+        expect(publisher_ats_applications_page.tab_panel.job_applications.count).to eq(0)
+      end
+    end
+  end
+
+  describe "transition: submitted to reviewed", :js do
+    let(:status) { "submitted" }
+
+    it "jobseeker and publisher can view job application" do
+      run_with_jobseeker(jobseeker) do
+        #
+        # jobseeker views all its applications
+        #
+        jobseeker_applications_page.load
+        job_applications_count = jobseeker.reload.job_applications.count
+        expect(jobseeker_applications_page.header).to have_text("Applications (#{job_applications_count})")
+        #
+        # view submitted application
+        #
+        jobseeker_applications_page.click_on_job_application(job_application.id)
+        expect(jobseeker_application_page).to be_displayed(id: job_application.id)
+        expect(jobseeker_application_page.tag).to have_text("submitted")
+      end
+
+      run_with_publisher(publisher) do
+        publisher_ats_applications_page.load(vacancy_id: vacancy.id)
+
+        #
+        # check tab all
+        #
+        publisher_ats_applications_page.select_tab(:tab_all)
+        expect(publisher_ats_applications_page.tab_panel.job_applications.count).to eq(1)
+
+        #
+        # check tab new
+        #
+        publisher_ats_applications_page.select_tab(:tab_submitted)
+        expect(publisher_ats_applications_page.tab_panel.job_applications.count).to eq(1)
+
+        display_status = publisher_ats_applications_page.tab_panel.job_applications.first.status
+        expect(display_status).to have_text("unread")
+
+        publisher_ats_applications_page.update_status(job_application) do |tag_page|
+          tag_page.select_and_submit("reviewed")
+        end
+
+        display_status = publisher_ats_applications_page.tab_panel.job_applications.first.status
+        expect(display_status).to have_text("reviewed")
+      end
+
+      run_with_jobseeker(jobseeker) do
+        #
+        # jobseeker views all its applications
+        #
+        jobseeker_applications_page.load
+        job_applications_count = jobseeker.reload.job_applications.count
+        expect(jobseeker_applications_page.header).to have_text("Applications (#{job_applications_count})")
+        #
+        # view submitted application
+        #
+        jobseeker_applications_page.click_on_job_application(job_application.id)
+        expect(jobseeker_application_page).to be_displayed(id: job_application.id)
+        expect(jobseeker_application_page.tag).to have_text("submitted") # TODO: should the jobseeker view reviewed instead of submitted here
+        expect(job_application.reload.status).to eq("reviewed")
+      end
+    end
+  end
+
+  describe "transition: reviewed to unsuccessful", :js do
+    let(:status) { "reviewed" }
+
+    it "jobseeker and publisher can view job application" do
```

Comment:
[Minor] We could change all of these test descriptions to describe what the publisher should see and what the jobseeker should see. For instance "publisher sees unsuccessful and jobseeker sees rejected tag" or something like that